docs/gc.md

```diff
@@ -141,15 +141,20 @@ Object ages are tracked for each region with a maximum of 24 possible generation
 
 ![The diagram is explained in the surrounding text](./cr/balanced_regions.png "Region structure and characteristics found in the object heap")
 
- The *eden* space is a set of regions of age 0, which contain the newest objects allocated. Regions of intermediate age serve as survivor regions to receive live objects moved from younger regions. When the eden space is full, a partial garbage collection (PGC) runs to reclaim space, typically by using a *copy forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available. The PGC is a stop-the-world operation, after which new regions are allocated to create a new eden space. A PGC always includes the eden space, but might include older regions. Objects within collectible regions of intermediate age N are evacuated to survivor regions of age N+1. Regions of age 24 are included in PGC collection sets in order to defragment them.
+The *eden* space is a set of regions of age 0, which contain the newest objects allocated. Regions of intermediate age or empty regions serve as survivor regions to receive live objects moved from younger regions. When the *eden* space is full, a partial GC cycle runs to reclaim space, typically by using a *copy forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available. The partial GC cycle is a *stop-the-world* (STW) operation, after which new regions are allocated to create a new *eden* space. A partial cycle always includes the *eden* space, but might include older regions. Objects from collectible regions of age N are moved into another region of the same age N or to an empty region that is assigned an age of N. Then, the ages of all regions across the heap are incremented by 1, except for the maximum age 24 regions. Regions of age 24 are included in partial GC collection sets in order to defragment them.
 
-Over time, transient objects drop out of the live set and viable survivor regions become scarce. This problem is solved by introducing a global mark phase (GMP) to mark and sweep the entire heap. GMP is a hybrid *incremental-concurrent* GC cycle. Each GMP increment is interleaved with a PGC cycle so that regions that are left sparse after sweeping the heap can be collected, cleared, and recycled during and after the GMP.
+Partial GC cycles work to reclaim free regions in the heap for allocating new objects. As much as 90% of this memory is reclaimed by moving objects out of *eden* regions to older regions.
```

Comment:
90% is just a typical ratio. 'as much as' sounds to me it's maximum

Reply:
As much as 90% of this memory is reclaimed by moving objects out of *eden* regions to older regions. 

and 

However, because some objects from *eden* regions always survive, 

are tightly couple sentences and possibly some rewording is needed for proper cause-consequence relationship. Because some Eden object survive (~10%), PGC cannot reclaim all Eden worth of region (but only 90%).

For example, to me it would be better to start the second sentence without however, but like:

Since some objects from eden always survive

Reply:
An alternative, which is more in line with your suggestion:

_Because some objects from *eden* regions always survive, a partial GC cycle can reclaim only about  90% of this memory. To keep up with object allocation, partial GC cycles also reclaim free regions by defragmenting older regions_.

Reply:
yes, that's perfect